src/main/java/org/folio/rest/helper/ExchangeRateHelper.java

```diff
@@ -0,0 +1,25 @@
+package org.folio.rest.helper;
+
+import io.vertx.core.Context;
+import org.folio.rest.acq.model.finance.ExchangeRate;
+
+import javax.money.convert.MonetaryConversions;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class ExchangeRateHelper extends AbstractHelper {
+  public ExchangeRateHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(okapiHeaders, ctx, lang);
+  }
+
+  public CompletableFuture<ExchangeRate> getExchangeRate(String from, String to) {
+    CompletableFuture<ExchangeRate> future = new CompletableFuture<>();
+    double exchangeRate = MonetaryConversions.getExchangeRateProvider().getExchangeRate(from, to).getFactor().doubleValue();
```

Comment:
Method getExchangeRate() can throw CurrencyConversionException if no such rate is available and MonetaryException if one of the currency codes passed is not valid. Maybe it makes sense to catch these runtime exceptions here and build corresponding HttpExceptions for codes 404 and 400? If yes, one need to improve raml file too.

Reply:
I see that you process these exception like one error 400 without additional detail. This isn't issue, just possible improvement of error's handling.

Reply:
I am not sure how ExchangeRateProvider works, but if It goes to third system, then all should be run in Vertx Future

Reply:
It is already in the Vertx future